src/matcher_combinators/parser.clj

```diff
@@ -38,6 +38,6 @@
                BigInt
                Character)
 
-(mimic-matcher matchers/contains-map IPersistentMap)
-(mimic-matcher matchers/equals-seq IPersistentVector)
-(mimic-matcher matchers/equals-seq IPersistentList)
+(mimic-matcher matchers/embeds IPersistentMap)
+(mimic-matcher matchers/equals IPersistentVector)
+(mimic-matcher matchers/equals IPersistentList)
```

Comment:
I might try to direct calls to the record creation functions like `->Value` instead of letting `equals` do it. Leaving it this way because I also might try to add some indirect to allow for an `match-exactly` that changes the default for maps to be `equals` instead of `embeds`.